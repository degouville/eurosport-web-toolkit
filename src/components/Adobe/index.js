import React from 'react';
import PropTypes from 'prop-types';

class AdobeClientInjector extends React.Component {
  constructor(props) {
    super(props);
    const { src } = this.props;
    this.adobeScript = document.createElement('script');
    this.adobeScript.id = 'adobe-script';
    this.adobeScript.src = src;
    this.adobeScript.async = false;

    this.satelliteScript = document.createElement('script');
    this.satelliteScript.id = 'satellite-script';
    this.satelliteScript.type = 'text/javascript';
    this.satelliteScript.async = false;
    this.satelliteScript.innerHTML = `if(typeof(_satellite) != 'undefined'){_satellite.pageBottom();}`;

    document.head.appendChild(this.adobeScript);
    document.body.appendChild(this.satelliteScript);
  }

  componentWillUnmount() {
    if (this.adobeScript) {
      document.head.removeChild(this.adobeScript);
    }
    if (this.satelliteScript) {
      document.body.removeChild(this.satelliteScript);
    }
  }

  render() {
    return null;
  }
}

const AdobeServerInjector = src => (
  <>
    <script id="adobe-script" src={src} />
    <script>
      {`if(typeof(_satellite) != 'undefined'){
             _satellite.pageBottom();}`}
    </script>
  </>
);

const Adobe = ({ src, isServerSide }) =>
  isServerSide ? <AdobeServerInjector src={src} /> : <AdobeClientInjector src={src} />;

Adobe.propTypes = {
  src: PropTypes.string.isRequired,
  isServerSide: PropTypes.bool.isRequired,
};

AdobeClientInjector.propTypes = {
  src: PropTypes.string.isRequired,
};

AdobeServerInjector.propTypes = {
  src: PropTypes.string.isRequired,
};

export default Adobe;
