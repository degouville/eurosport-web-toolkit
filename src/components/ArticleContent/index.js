import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import * as breakpoints from '../../breakpoints';
import * as colors from '../../colors';
import { Row, Column } from '../../elements/Grid';

const StyledWrapper = styled(Row)`
  color: ${colors.mirage};
`;

export const StyledTeaser = styled.h4`
  font-size: 16px;
  font-weight: bold;
  line-height: 24px;
  ${breakpoints.wide(css`
    font-size: 18px;
    line-height: 28px;
  `)};
`;

export const StyledParagraphs = styled.div`
  font-size: 14px;
  margin-top: 25px;
  line-height: 20px;
  ${breakpoints.wide(css`
    font-size: 16px;
    line-height: 24px;
  `)};
`;

const StyledParagraph = styled.p`
  margin-top: 15px;
`;

const ArticleContent = props => {
  const { teaser, paragraphs, ...otherProps } = props;
  return (
    <StyledWrapper {...otherProps}>
      <Column mediumOffset="2" medium="8" large="5">
        {teaser && <StyledTeaser>{teaser}</StyledTeaser>}
        {paragraphs.length > 0 && (
          <StyledParagraphs>
            {paragraphs.map((paragraph, index) => (
              // eslint-disable-next-line
              <StyledParagraph key={index}>{paragraph}</StyledParagraph>
            ))}
          </StyledParagraphs>
        )}
      </Column>
    </StyledWrapper>
  );
};

ArticleContent.defaultProps = {
  teaser: '',
  paragraphs: [],
};

ArticleContent.propTypes = {
  teaser: PropTypes.string,
  paragraphs: PropTypes.arrayOf(PropTypes.string),
};

export default ArticleContent;
