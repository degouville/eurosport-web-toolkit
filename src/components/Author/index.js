import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { css } from '@emotion/core';
import E from '../../assets/channels/E.svg';
import * as breakpoints from '../../breakpoints';
import * as colors from '../../colors';

const StyledAuthor = styled.div`
  display: flex;
  align-items: center;
`;

const StyledImage = styled.img`
  height: 32px;
  width: 32px;
  ${breakpoints.small(css`
    height: 40px;
    width: 40px;
  `)};
`;

const StyledSpacer = styled.div`
  width: 1px;
  height: 48px;
  background: ${colors.coreLightMinus1};
  opacity: 0.3;
  margin: 0 18px 0 11px;
  ${breakpoints.small(css`
    height: 64px;
    margin: 0 17px 0 13px;
  `)};
  ${breakpoints.small(css`
    height: 56px;
    margin: 0 33px 0 26px;
  `)};
`;

const StyledContent = styled.div`
  flex: 1;
  font-size: 14px;
  line-height: 24px;
  ${breakpoints.small(css`
    font-size: 16px;
  `)};
`;

const StyledName = styled.div`
  color: ${colors.coreLightMinus1};
  font-weight: bold;
`;

const StyledTime = styled.div`
  color: ${colors.coreLightMinus1};
`;

const Author = ({ name, img, time, ...props }) => {
  const avatar = img || E;
  return (
    <StyledAuthor {...props}>
      <StyledImage src={avatar} alt={name} />
      <StyledSpacer />
      <StyledContent>
        <StyledName>{name}</StyledName>
        <StyledTime>{time}</StyledTime>
      </StyledContent>
    </StyledAuthor>
  );
};

Author.propTypes = {
  name: PropTypes.string.isRequired,
  time: PropTypes.string.isRequired,
  img: PropTypes.string,
};

Author.defaultProps = {
  img: E,
};

export default Author;
