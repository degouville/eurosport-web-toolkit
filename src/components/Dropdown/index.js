import React from 'react';
import PropTypes from 'prop-types';
import styled, { css } from 'react-emotion';
import Chevron from 'src/assets/chevron.component.svg';
import { small, medium } from 'src/breakpoints';
import { H3, H5 } from 'src/typography';
import { removeUnit } from './utils';

export const StyledContainer = styled.div`
  display: flex;
  align-items: center;
  cursor: pointer;
`;

export const StyledOverlay = styled.div`
  position: fixed;
  width: 100vw;
  height: 100vh;
  top: 0;
  left: 0;
  z-index: 1;
  display: none;

  ${({ isDropdownOpen }) =>
    isDropdownOpen &&
    css`
      display: block;
    `}
`;

export const StyledWrapper = styled.div`
  position: relative;

  ${medium(css`
    width: fit-content;
  `)};
`;

export const StyledText = styled(H3)`
  color: ${({ theme }) => theme.dropdown.label.color};
  text-transform: uppercase;

  ${medium(css`
    display: flex;
    align-items: center;
  `)}
`;

export const StyledArrow = styled(Chevron)`
  height: 12px;
  width: 12px;
  transform: rotate(90deg);
  margin-left: 5px;
  margin-top: 2px;
  path {
    fill: ${({ theme }) => theme.dropdown.label.color};
  }
`;

const StyledHeading = styled(H5)`
  color: ${({ theme }) => theme.dropdown.heading.color};
  margin-bottom: 15px;
  text-transform: uppercase;
`;

const StyledDropdownTop = () => css`
  top: auto;
  bottom: 42px;

  &::before {
    top: auto;
    transform: translateX(-50%) rotate(220deg);
    bottom: -6px;
  }

  ${small(css`
    top: auto;
    bottom: 42px;
  `)}
`;

const StyledDropdownWithCustomWidth = ({ width }) => {
  const outerWidth = removeUnit(width) + 16.25;

  return css`
    width: ${width};

    ${medium(css`
      right: -${outerWidth / 2}px;
      @-moz-document url-prefix() {
        right: -${outerWidth / 2}px;
      }
    `)}
  `;
};

export const StyledDropdown = styled.div`
  position: absolute;
  width: calc(100% - 25px);
  top: 42px;
  display: none;
  z-index: 2;
  opacity: 0.95;
  padding: 8px;
  align-self: flex-end;
  border: solid 1px ${({ theme }) => theme.dropdown.border};
  border-radius: 3px;
  background-color: ${({ theme }) => theme.dropdown.background};

  &::before {
    content: '';
    position: absolute;
    width: 10px;
    border-top: 1px solid ${({ theme }) => theme.dropdown.border};
    border-left: 1px solid ${({ theme }) => theme.dropdown.border};
    border-radius: 1px;
    background: ${({ theme }) => theme.dropdown.background};
    height: 10px;
    display: block;
    margin-left: 50%;
    transform: translateX(-50%) rotate(45deg);
    top: -6px;
    left: 0;
  }

  ${medium(css`
    padding: 17px 8px 12px 17px;
    right: -183px;
    @-moz-document url-prefix() {
      right: -183px;
    }
    @supports (-ms-ime-align: auto) {
      right: unset;
      width: auto;
    }
    width: fit-content;
  `)};

  ${({ isDropdownOpen }) =>
    isDropdownOpen &&
    css`
      display: block;
    `}

  ${({ isTop }) => isTop && StyledDropdownTop};
  ${({ width }) => width && StyledDropdownWithCustomWidth};
`;

const StyledUl = styled.ul`
  max-height: 130px;
  overflow-y: auto;
  overflow-x: hidden;
  scrollbar-width: thin;
  scrollbar-color: ${({ theme }) => `${theme.scrollbar.color} ${theme.scrollbar.background}`};

  ::-webkit-scrollbar {
    width: 6px;
  }

  ::-webkit-scrollbar-track {
    background: ${({ theme }) => theme.scrollbar.background};
    border-radius: 2px;
  }

  ::-webkit-scrollbar-thumb {
    background: ${({ theme }) => theme.scrollbar.color};
    border-radius: 2px;
  }

  ${({ width }) =>
    medium(css`
      width: ${width || '350px'};
    `)};
`;

export const StyledLi = styled.li`
  color: ${({ isSelected, theme }) =>
    isSelected ? theme.dropdown.list.selected.color : theme.dropdown.list.default.color};
  font-size: 14px;
  cursor: pointer;
  padding: 8px 0;
  margin-right: 10px;
  display: flex;
  justify-content: space-between;
  align-items: center;

  &:first-of-type {
    padding-top: 0;
  }
  &:not(:last-of-type) {
    border-bottom: solid 1px ${({ theme }) => theme.dropdown.border};
  }
  &:hover {
    color: ${({ theme }) => theme.dropdown.list.selected.color};
  }

  ${small(css`
    padding: 13px 0;
  `)}
`;

export const StyledCheckMark = styled.span`
  color: ${({ theme }) => theme.dropdown.checkMark.color};
  margin: 0 0 0 1em;
`;

const StyledP = styled.p`
  width: 100%;
`;

class Dropdown extends React.Component {
  state = {
    isDropdownOpen: false,
    selectedOption: null,
  };

  componentWillMount() {
    this.initSelectedOption();
  }

  initSelectedOption = () => {
    const { options, initialOptionID } = this.props;
    const matchInitialID = initialID => ({ id }) => id === initialID;
    const initialOption = options.find(matchInitialID(initialOptionID));
    const selectedOption = initialOption || options[0];
    this.setState({ selectedOption });
  };

  onOptionClick = id => {
    const { options, onItemSelected } = this.props;
    const matchingOption = options.find(option => option.id === id);
    onItemSelected && onItemSelected(matchingOption);

    if (matchingOption) {
      this.setState({ selectedOption: matchingOption, isDropdownOpen: false });
    }
  };

  toggleOpen = () => {
    // eslint-disable-next-line no-unused-vars
    this.setState(state => ({
      isDropdownOpen: !state.isDropdownOpen,
    }));
  };

  render() {
    const { children, className, heading, isTop, options, width } = this.props;
    const { isDropdownOpen, selectedOption } = this.state;

    return (
      <StyledWrapper>
        <StyledOverlay isDropdownOpen={isDropdownOpen} onClick={this.toggleOpen} />
        <StyledDropdown className={className} isDropdownOpen={isDropdownOpen} isTop={isTop} width={width}>
          {heading && <StyledHeading>{heading}</StyledHeading>}
          <StyledUl width={width}>
            {options &&
              options.map(option => (
                <StyledLi
                  onClick={() => this.onOptionClick(option.id)}
                  isSelected={selectedOption.id === option.id}
                  key={option.id}
                >
                  <StyledP>{option.text}</StyledP>
                  {selectedOption.id === option.id && <StyledCheckMark> âœ“</StyledCheckMark>}
                </StyledLi>
              ))}
          </StyledUl>
        </StyledDropdown>
        <StyledContainer onClick={this.toggleOpen}>
          <StyledText>
            {children || (
              <>
                {selectedOption.text}
                <StyledArrow />
              </>
            )}
          </StyledText>
        </StyledContainer>
      </StyledWrapper>
    );
  }
}

Dropdown.defaultProps = {
  children: null,
  className: undefined,
  heading: null,
  isTop: false,
  onItemSelected: null,
  initialOptionID: null,
  width: null,
};

Dropdown.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  heading: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  isTop: PropTypes.bool,
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      text: PropTypes.string,
    })
  ).isRequired,
  initialOptionID: PropTypes.number,
  onItemSelected: PropTypes.func,
  width: PropTypes.string,
};

export default Dropdown;
