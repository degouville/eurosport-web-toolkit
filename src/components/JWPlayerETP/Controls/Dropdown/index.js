import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';

const Dropdown = ({ children, bottomDisplay }) => (
  <MainContainer>
    <StyledContainer bottomDisplay={bottomDisplay}>
      <StyledDropdownPick bottomDisplay={bottomDisplay} />
      <StyledDropdown>{children}</StyledDropdown>
    </StyledContainer>
  </MainContainer>
);

Dropdown.propTypes = {
  children: PropTypes.node.isRequired,
  bottomDisplay: PropTypes.bool.isRequired,
};

const MainContainer = styled.div`
  position: relative;
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
`;

const StyledContainer = styled.div`
  position: absolute;
  display: flex;
  flex: 1;
  justify-content: center;
  align-items: center;
  flex-direction: ${({ bottomDisplay }) => (bottomDisplay ? 'column-reverse' : 'column')};
  ${({ bottomDisplay }) => bottomDisplay && 'bottom: 100%;'}
`;

const StyledDropdownPick = styled.div`
  border: solid 1px ${({ theme }) => theme.playerControls.dropdown.border};
  ${({ bottomDisplay }) => bottomDisplay && 'border-left: none;'}
  ${({ bottomDisplay }) => bottomDisplay && 'border-top: none;'}
  ${({ bottomDisplay }) => !bottomDisplay && 'border-right: none;'}
  ${({ bottomDisplay }) => !bottomDisplay && 'border-bottom: none;'}
   
  border-radius: 3px;
  background-color: ${({ theme }) => theme.playerControls.dropdown.background};
  height: 10px;
  width: 10px;
  transform: ${({ bottomDisplay }) => (bottomDisplay ? ' translate(0%, -50%)' : 'translate(0%, 50%)')} rotate(45deg);
`;

export const StyledDropdown = styled.div`
  border: solid 1px ${({ theme }) => theme.playerControls.dropdown.border};
  border-radius: 3px;
  background-color: ${({ theme }) => theme.playerControls.dropdown.background};
  padding: 18px;
`;

export default Dropdown;
