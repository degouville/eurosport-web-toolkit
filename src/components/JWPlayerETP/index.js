import React from 'react';
import PropTypes from 'prop-types';
import ETPlayerComponent from '@eurosport/toolkit-player';

const ETPlayer = ({
  prefLang,
  elementId,
  jwplayerData,
  videoData,
  onReady,
  onPlay,
  onPause,
  onAdBreakStart,
  onAdBreakComplete,
  onAdStart,
  onAdTime,
  onError,
  onLoginModalReady,
  onLoginModalDone,
}) => (
  <ETPlayerComponent
    locale={prefLang}
    elementId={elementId}
    jwplayerData={jwplayerData}
    videoData={videoData}
    onReady={onReady}
    onPlay={onPlay}
    onPause={onPause}
    onAdBreakStart={onAdBreakStart}
    onAdBreakComplete={onAdBreakComplete}
    onAdStart={onAdStart}
    onAdTime={onAdTime}
    onError={onError}
    onLoginModalReady={onLoginModalReady}
    onLoginModalDone={onLoginModalDone}
  />
);

ETPlayer.defaultProps = {
  onReady: null,
  onPlay: null,
  onPause: null,
  onAdBreakStart: null,
  onAdBreakComplete: null,
  onAdStart: null,
  onAdTime: null,
  onError: null,
  onLoginModalReady: null,
  onLoginModalDone: null,
};

ETPlayer.propTypes = {
  prefLang: PropTypes.string.isRequired,
  elementId: PropTypes.string.isRequired,

  jwplayerData: PropTypes.shape({
    id: PropTypes.string.isRequired,
    key: PropTypes.string.isRequired,
  }).isRequired,

  videoData: PropTypes.shape({
    provider: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    url: PropTypes.string,
    title: PropTypes.string,
    duration: PropTypes.number,
    type: PropTypes.string,

    sonic: PropTypes.shape({
      baseUrl: PropTypes.string.isRequired,
      realm: PropTypes.string.isRequired,
    }),

    mux: PropTypes.shape({
      env_key: PropTypes.string.isRequired,
    }),

    freewheelAdParams: PropTypes.shape({
      adManagerURL: PropTypes.string.isRequired,
      serverURL: PropTypes.string.isRequired,
      networkID: PropTypes.number.isRequired,
      fwassetId: PropTypes.string,
      profileId: PropTypes.string,
      sectionId: PropTypes.string,
      video_targeting: PropTypes.string,
      auth: PropTypes.string,
      sport: PropTypes.string,
      afid: PropTypes.string,
      vdur: PropTypes.number,
      _fw_gdpr: PropTypes.number,
      _fw_gdpr_consent: PropTypes.string,
    }),
  }).isRequired,

  // Events
  onReady: PropTypes.func,
  onPlay: PropTypes.func,
  onPause: PropTypes.func,
  onAdBreakStart: PropTypes.func,
  onAdBreakComplete: PropTypes.func,
  onAdStart: PropTypes.func,
  onAdTime: PropTypes.func,
  onError: PropTypes.func,
  onLoginModalReady: PropTypes.func,
  onLoginModalDone: PropTypes.func,
};

export default ETPlayer;
