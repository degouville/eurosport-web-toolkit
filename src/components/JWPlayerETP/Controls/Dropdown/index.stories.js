import React from 'react';
import { storiesOf } from '@storybook/react';
import theme from 'src/theme';
import { boolean } from '@storybook/addon-knobs';
import styled from 'react-emotion';
import Dropdown from './index';
import Slider from '../Slider';

const dropdownStories = storiesOf('Components|PlayerDropdown', module);

dropdownStories.add('Default', () => (
  <MainContainer>
    <Dropdown theme={theme.playerControls.dropdown} bottomDisplay={boolean('bottomDisplay', true)}>
      <SliderContainer>
        <Slider max={100} onChange={() => null} min={0} value={50} defaultValue={0} {...theme.playerControls.volume} />
      </SliderContainer>
    </Dropdown>
  </MainContainer>
));

const MainContainer = styled.div`
  margin-top: 200px;
  display: flex;
  flex: 1;
  flex-direction: column;
  align-items: center;
`;

const SliderContainer = styled.div`
  height: 100px;
`;
