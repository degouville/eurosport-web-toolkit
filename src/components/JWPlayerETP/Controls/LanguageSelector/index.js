import React, { useCallback, useMemo } from 'react';
import styled, { css } from 'react-emotion';
import PropTypes from 'prop-types';
import Dropdown from 'src/components/Dropdown';
import languageIcon from 'src/assets/icon-language.svg';
import { audioTracks as audioTracksPropTypes } from 'src/types';
import { small } from 'src/breakpoints';
import Icon from '../UI/icon';
import useScreenInformation from '../useScreenInformation';

const StyledDropdown = styled(Dropdown)`
  :before {
    left: auto;
    margin-left: auto;
    right: 5px;
  }

  ${small(css`
    bottom: 58px;
    right: 0;
  `)}
`;

const StyledDropdownForMobile = styled(Dropdown)`
  bottom: auto;
  right: 0;
  top: 30px;
  width: 150px;

  :before {
    left: auto;
    margin-left: auto;
    right: 0;
  }

  ul {
    height: 100px;
  }
`;

export const LanguageSelector = ({ audioTracks, onAudioTrackChange }) => {
  const { isMobile } = useScreenInformation();

  const onLanguageSelected = useCallback(({ id }) => onAudioTrackChange(id), [onAudioTrackChange]);

  const options = useMemo(
    () =>
      audioTracks &&
      audioTracks.map(({ name }, index) => ({
        id: index,
        text: name,
      })),
    [audioTracks]
  );

  return isMobile ? (
    <StyledDropdownForMobile InitialOptionID={0} onItemSelected={onLanguageSelected} options={options}>
      <Icon alt="Language" src={languageIcon} />
    </StyledDropdownForMobile>
  ) : (
    <StyledDropdown
      heading="audio"
      InitialOptionID={0}
      isTop
      onItemSelected={onLanguageSelected}
      options={options}
      width="200px"
    >
      <Icon alt="Language" src={languageIcon} />
    </StyledDropdown>
  );
};

LanguageSelector.propTypes = {
  audioTracks: audioTracksPropTypes.isRequired,
  onAudioTrackChange: PropTypes.func.isRequired,
};

export default LanguageSelector;
