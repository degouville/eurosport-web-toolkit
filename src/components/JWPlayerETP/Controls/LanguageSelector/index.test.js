import React from 'react';
import { shallow } from 'enzyme';
import { StyledLi } from 'src/components/Dropdown';
import LanguageSelector from './index';
import audioTracksMock from './mocks/audio-tracks.json';

describe('LanguageSelector', () => {
  const createDefaultProps = newProps => ({
    audioTracks: [...audioTracksMock],
    onAudioTrackChange: () => {},
    ...newProps,
  });

  describe('Snapshots', () => {
    it('Should match snapshot', () => {
      // Given
      const props = createDefaultProps();
      const wrapper = shallow(<LanguageSelector {...props} />);

      // Expect
      expect(wrapper).toMatchSnapshot();
    });
  });

  it('Should display all 3 languages', () => {
    // Given
    const props = createDefaultProps();
    const wrapper = shallow(<LanguageSelector {...props} />).dive();
    const languages = wrapper.dive().find(StyledLi);

    // Expect
    expect(languages).toHaveLength(3);
  });

  it('Should change selected language to bulgarian language', () => {
    // Given
    const props = createDefaultProps({
      onAudioTrackChange: jest.fn(),
    });
    const wrapper = shallow(<LanguageSelector {...props} />).dive();
    const languages = wrapper.dive().find(StyledLi);
    const bulgarianLanguage = languages.at(2);

    // When
    bulgarianLanguage.simulate('click');

    // Expect
    expect(props.onAudioTrackChange).toHaveBeenCalledWith(2);
  });
});
