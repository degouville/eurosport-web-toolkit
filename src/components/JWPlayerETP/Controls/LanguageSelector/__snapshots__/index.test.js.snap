// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageSelector Snapshots Should match snapshot 1`] = `
<Styled(Dropdown)
  InitialOptionID={0}
  heading="audio"
  initialOptionID={null}
  isTop={true}
  onItemSelected={[Function]}
  options={
    Array [
      Object {
        "id": 0,
        "text": "English",
      },
      Object {
        "id": 1,
        "text": "Greek",
      },
      Object {
        "id": 2,
        "text": "Bulgarian",
      },
    ]
  }
  width="200px"
>
  <Styled(img)
    alt="Language"
    src="/eurosport-web-toolkit/203b0a9c38be6e5e8480dfe9c61b5a34.svg"
  />
</Styled(Dropdown)>
`;
